frontend/public/components/utils/list-dropdown.jsx

```diff
@@ -177,10 +187,56 @@ ListDropdown.propTypes = {
 
 const NsDropdown_ = (props) => {
   const openshiftFlag = props.flags[FLAGS.OPENSHIFT];
+
   if (flagPending(openshiftFlag)) {
     return null;
   }
-  const kind = openshiftFlag ? 'Project' : 'Namespace';
+
+  const nsModel = modelFor('Namespace');
+  const useNamespaceAccessReview = () => {
+    const canCreateNamespace = useAccessReview({
+      group: nsModel.apiGroup,
+      resource: nsModel.plural,
+      verb: 'create',
+    });
+    return canCreateNamespace;
+  };
+
+  const prModel = modelFor('Project');
+  const useProjectAccessReview = () => {
+    const canCreateProject = useAccessReview({
+      group: prModel.apiGroup,
+      resource: prModel.plural,
+      verb: 'create',
+    });
+    return canCreateProject;
+  };
+
+  const actionItems = useNamespaceAccessReview
+    ? [
+        {
+          actionTitle: `Create ${nsModel.label}`,
+          actionKey: CREATE_NEW_NAMESPACE,
+        },
+      ]
+    : useProjectAccessReview
+    ? [
+        {
+          actionTitle: `Create ${prModel.label}`,
+          actionKey: CREATE_NEW_PROJECT,
+        },
+      ]
+    : [];
```

Comment:
If we establish the model for the resource the user can create as suggested above, we shouldn't need any condition here. We can just build the action key from the model.label as well. So the end result would be something like:

```tsx
  const actionItems = model ? [
    {
      actionTitle: `Create ${model.label}`,
      actionKey: `CREATE_${model.label}`,
    }
  ]
```

Reply:
You would also have to update your onChange function to use the new actionKeys.